스터디 그룹 멤버 강퇴 API

kimshineday opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #211 
- 작업 요약: 스터디 그룹 멤버 강퇴 API 기능 구현

## 📄 상세 내용
- [ ] 스터디 그룹 리더 검증 (스터디 그룹 리더 퍼미션 활용)
- [ ] 선택한 그룹원이 같은 그룹원인지 확인 필요
- [ ] GroupMember 에서 해당 데이터 삭제
  - [ ] 스터디 그룹 탈퇴에서 사용하는 삭제 코드 사용 (함수 이름 변경 groupmember_delete_service)
- [ ] test code 작성

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

kimshineday committed:
♻️ refactor: 스터디 그룹 멤버 삭제 코드 통일

- 스터디 그룹 멤버 탈퇴 및 강퇴 요청시 삭제 로직 통일

kimshineday committed:
✨ feat: 스터디 그룹 멤버 강퇴 기능 구현 (view, url)

- url 추가
- view 코드 작성

kimshineday committed:
✅ test: 스터디 그룹 멤버 강퇴 API 단위 테스트 작성

- 실패 케이스
  - 리더 권한이 없는 경우
  - 리더 자기 자신을 강퇴하려는 경우
  - 강퇴하려는 사람이 스터디 그룹원이 아닌 경우
- 성공 케이스

SimJunSik commented on apps/studies/services/delete_member_service.py:
```diff
@@ -20,9 +20,9 @@ def check_leader(self) -> bool:
 
-    def studygroup_withdraw_service(self) -> None:
+    def groupmember_delete_service(self) -> None:
         """
-        스터디 그룹 탈퇴 기능
+        스터디 그룹 탈퇴 및 강퇴 기능
```
탈퇴 서비스랑 강퇴 서비스를 나누는 건 어떨까요?
`self.group.members.remove(self.target)` 가 공통된 부분이니까 지금처럼 이걸 일반화 시킬 수도 있겠지만,
탈퇴랑 강퇴는 엄연히 다른 기능이기는 해서요

logging 을 할 때도 달라질 수 있고, 이후에 각 기능이 조금씩 달라질 수도 있으니 미리 나눠서 명확하게 하는 걸 추천드립니다

바로 아래에 있는 코드만 봐도 탈퇴 기능을 위한 메세지로 보이네요
```python
if self.check_leader():
   raise PermissionDenied("그룹 리더는 리더 위임 후 그룹을 탈퇴할 수 있습니다.")
```

kimshineday replied:
중복 코드가 되어버려 고민하다 하나로 합쳤는데 아무래도 분리하는게 더 좋아보이긴 합니다..!
강퇴 관련 메세지와 함께 분리 하겠습니다!
감사합니다!

kimshineday committed:
♻️ refactor: 스터디 그룹 탈퇴, 멤버 강퇴 서비스 코드 분리

- 탈퇴, 강퇴가 서로 다른 기능이므로 코드 분리
  - 탈퇴 : withdraw_studygroup
  - 강퇴 : kick_groupmember
- 서비스 코드 수정으로 인한 view 코드 수정
